execution/function/scalar.go

```diff
@@ -6,15 +6,32 @@ package function
 import (
 	"context"
 	"math"
+	"time"
 
 	"github.com/prometheus/prometheus/model/labels"
 
 	"github.com/thanos-io/promql-engine/execution/model"
 )
 
 type scalarFunctionOperator struct {
-	pool *model.VectorPool
-	next model.VectorOperator
+	pool     *model.VectorPool
+	next     model.VectorOperator
+	ti       model.TimingInformation
+	duration time.Duration
+}
+
+func (o *scalarFunctionOperator) Analyze() (*model.TimingInformation, []model.ObservableVectorOperator) {
+
+	// Create a TimingInformation instance and add CPU time taken
+	timingInfo := &model.TimingInformation{}
+	timingInfo.AddCPUTimeTaken(o.duration)
+
+	// Create a slice to store any observable vector operators found during analysis
+	observableOperators := make([]model.ObservableVectorOperator, 0)
```

Comment:
Not quite. We aren't really finding  The semantics for this should be same as `Explain`, that is, it should return any next operators it can find.